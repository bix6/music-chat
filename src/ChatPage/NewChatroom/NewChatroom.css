.new-chatroom-div {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-image: linear-gradient(135deg, #fccf31 10%, #f55555 100%);
  display: flex;
  align-items: center;
}

.new-chatroom-form {
  border: 2px solid #9708cc;
  background-image: linear-gradient(20deg, #9708cc 10%, #43cbff 100%);
  max-width: 500px;
  margin: 1rem auto;
  display: flex;
  flex-direction: column;
  text-align: center;
  padding: 2rem 4rem;
}

.new-chatroom-form input {
  margin: 0.3rem 0 1rem 0;
}

.new-chatroom-form button {
  margin: 0.5rem auto;
  max-width: 90%;
}
