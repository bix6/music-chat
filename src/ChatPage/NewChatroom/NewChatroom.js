import React from "react";
import DisplayError from "../../shared-components/DisplayError/DisplayError";

class NewChatroom extends React.Component {
  state = {
    newChatroom: "",
    chatroomTouched: false,
    newDescription: "",
    descriptionTouched: false,
    successMessage: "",
  };

  updateNewChatroom = (chatroom) => {
    this.setState({
      newChatroom: chatroom,
      chatroomTouched: true,
    });
  };

  validateNewChatroom = () => {
    const chatroom = this.state.newChatroom.trim();

    if (chatroom.length < 1) {
      return "Chatroom must be 1 character or more";
    }
  };

  updateNewChatroomDescription = (description) => {
    this.setState({
      newDescription: description,
      descriptionTouched: true,
    });
  };

  validateNewChatroomDescription = () => {
    const description = this.state.newDescription.trim();

    if (description.length < 1) {
      return "Description must be 1 character or more";
    }
  };

  handleSubmit = (e) => {
    e.preventDefault();
    this.props.createChatroom(
      this.state.newChatroom,
      this.state.newDescription
    );
    this.setState({
      newChatroom: "",
      chatroomTouched: false,
      newDescription: "",
      descriptionTouched: false,
      successMessage: "Chatroom Created",
    });
    setTimeout(() => {
      this.setState({
        successMessage: "",
      });
    }, 3000);
  };

  render() {
    return (
      <form
        className="new-chatroom-form"
        onSubmit={(e) => this.handleSubmit(e)}
      >
        <label htmlFor="new-chatroom">New Chatroom:</label>
        <input
          type="text"
          name="new-chatroom"
          id="new-chatroom"
          value={this.state.newChatroom}
          onChange={(e) => this.updateNewChatroom(e.target.value)}
        />
        <label htmlFor="new-chatroom-description">Description:</label>
        <input
          type="text"
          name="new-chatroom-description"
          id="new-chatroom-description"
          value={this.state.newDescription}
          onChange={(e) => this.updateNewChatroomDescription(e.target.value)}
        />
        <button
          type="submit"
          disabled={
            this.validateNewChatroom() || this.validateNewChatroomDescription()
          }
        >
          Create Chatroom
        </button>
        {this.state.chatroomTouched && (
          <DisplayError message={this.validateNewChatroom()} />
        )}
        {this.state.descriptionTouched && (
          <DisplayError message={this.validateNewChatroomDescription()} />
        )}
        {!!this.state.successMessage && <div>{this.state.successMessage}</div>}
      </form>
    );
  }
}

export default NewChatroom;
