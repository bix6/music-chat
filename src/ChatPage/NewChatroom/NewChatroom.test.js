import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";
import NewChatroom from "./NewChatroom";

describe("NewChatroom Component", () => {
  it("Smoke Test: Renders Default", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <NewChatroom />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Snapshot Test: Default", () => {
    const wrapper = shallow(<NewChatroom />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
