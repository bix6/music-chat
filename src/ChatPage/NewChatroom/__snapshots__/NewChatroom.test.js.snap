// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewChatroom Component Snapshot Test: Default 1`] = `
<div
  className="new-chatroom-div"
>
  <form
    className="new-chatroom-form"
    onSubmit={[Function]}
    style={
      Object {
        "display": undefined,
      }
    }
  >
    <label
      htmlFor="new-chatroom"
    >
      New Chatroom:
    </label>
    <input
      id="new-chatroom"
      name="new-chatroom"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="new-chatroom-description"
    >
      Description:
    </label>
    <input
      id="new-chatroom-description"
      name="new-chatroom-description"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      disabled="Chatroom must be 1 character or more"
      type="submit"
    >
      Create Chatroom
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </form>
</div>
`;
